recipe_scrapers/schoolofwok.py

```diff
@@ -0,0 +1,172 @@
+from collections import defaultdict
+
+from ._abstract import AbstractScraper
+from ._grouping_utils import IngredientGroup
+from ._utils import normalize_string
+
+
+class SchoolOfWok(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "schoolofwok.co.uk"
+
+    def author(self):
+        return "School of Wok"
+
+    def site_name(self):
+        return "School of Wok"
+
+    def title(self):
+        return self.soup.find("h1").get_text()
+
+    def cuisine(self):
+        categoryheader = self.soup.find(
+            string=lambda text: text and text.lower() == "cuisine"
+        )
+
+        if not categoryheader:
+            return ""
+
+        return categoryheader.find_next("p").get_text()
+
+    def total_time(self):
+        timeheader = self.soup.find(string=lambda text: text and text.lower() == "time")
+
+        if not timeheader:
+            return ""
+
+        return int(timeheader.find_next("p").get_text().split(" ")[0])
+
+    def yields(self):
+        servingheader = self.soup.find(
+            string=lambda text: text and text.lower() == "servings"
+        )
+
+        if not servingheader:
+            return ""
+
+        servingsize = servingheader.find_next("p").get_text().split(" ")[0]
+        return (
+            f"{servingsize} serving"
+            if int(servingsize) == 1
+            else f"{servingsize} servings"
+        )
+
+    def image(self):
+        return self.soup.find("img", {"alt": "recipe"}).get("src")
+
+    def ingredients(self):
+        possibleingredients = (
+            self.soup.find("section", {"id": "recipe-ingredients"})
+            .findChildren(
+                "div",
+                {
+                    "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                },
+            )[0]
+            .find_all("p")
+        )
+        ingredientslist = []
+
+        if possibleingredients:
+            for ingredient in possibleingredients:
+                if not ingredient.find("strong") and normalize_string(
+                    ingredient.get_text()
+                ):
+                    ingredientslist.append(normalize_string(ingredient.get_text()))
+        else:
+            possibleingredients = (
+                self.soup.find("section", {"id": "recipe-ingredients"})
+                .findChildren(
+                    "div",
+                    {
+                        "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                    },
+                )[0]
+                .find_all("li")
+            )
+            ingredientslist = [
+                normalize_string(ingredient.get_text())
+                for ingredient in possibleingredients
+            ]
+
+        return ingredientslist
```

Comment:
```suggestion
        section = self.soup.find("section", {"id": "recipe-ingredients"})
        div_class = "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
        divs = section.findChildren("div", {"class": div_class})

        if not divs:
            return []

        paragraphs = divs[0].find_all("p")
        ingredients_list = []

        if paragraphs:
            for paragraph in paragraphs:
                text = normalize_string(paragraph.get_text())
                if text and not paragraph.find("strong"):
                    ingredients_list.append(text)
        else:
            list_items = divs[0].find_all("li")
            ingredients_list = [
                normalize_string(item.get_text()) for item in list_items
            ]

        return ingredients_list
```

This is largely a readability refactor but it does also add specific handling for an instance where the div does not exist & consolidates a little duplicate code.

Reply:
Thank you! This definitely improved the readability of the code instead of having nested if statements.